src/movies.js

```diff
@@ -1,31 +1,141 @@
+    //   duration: '2h 7min',
+    //   genre: ['Action', 'Adventure', 'Fantasy'],
+    //   score: 8.3
+    // },
+  });
+  let listOfDrama = listOfSpielberg.filter(function (element) {
+    for (let i = 0; i < element['genre'].length; i++) {
+      let current_genre = element['genre'][i];
+      //return current_genre === "Drama";
+      if (current_genre === 'Drama') {
+        return true;
+      }
+    }
+    return false;
+  });
+  return listOfDrama.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(listOfmovies) {
+  if (listOfmovies.length === 0) {
+    return 0;
+  }
+  // I want all the scores first
+  let listOfScores = listOfmovies.map(function (element) {
+    return element['score'];
+    // I want to conculatethe average score
+  });
+  let listOfFilterdScore = listOfScores.filter(function (number) {
+    return number; // checking if a number is 'true'
+    // where are filtering movies that don't have a score
+  });
+  let sumScores = listOfFilterdScore.reduce(function (sum, number) {
+    return (sum += number);
+  }, 0);
+  // I am having a total amount of moviescore and deviding that bye the amount of movies
+  return parseFloat((sumScores / listOfmovies.length).toFixed(2));
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(listOfmovies) {
+  let listofmoviesWithGenre = listOfmovies.filter(function (element) {
+    return element['genre'];
+  });
+
+  // cheak if these movies have drama
+  let listOfDrama = listofmoviesWithGenre.filter(function (movie) {
+    // i need to look through the list of genre and cheak if it containts drama
+    for (i = 0; i < movie['genre'].length; i++) {
+      if (movie['genre'][i] === 'Drama') {
+        return true;
+      }
+    }
+    return false;
+  });
+  if (listOfDrama.length === 0) {
+    return 0;
+  }
+
+  let sumScores = listOfDrama.reduce(function (sum, movie) {
+    return (sum += movie['score']);
+  }, 0);
+  return parseFloat((sumScores / listOfDrama.length).toFixed(2));
+  // couculate the avarage
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(listOfmovies) {
+  //console.log(listOfmovies);
+  listCopy = listOfmovies.map(function (e) {
+    return e;
+  });
+  listCopy.sort(function (a, b) {
+    let decision = a['year'] - b['year'];
+    if (decision === 0) {
+      return a['title'].localeCompare(b['title']);
+    } else {
+      return decision;
+    }
+  });
+  //console.log(listCopy);
+  return listCopy;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically() {}
+function orderAlphabetically(listOfmovies) {
+  //sort all the movie titles
+  
+  listOfmovies.sort(function (a, b) {
+    return a['title'].localeCompare(b['title']);
+  });
```

Comment:
Task asks you not to mutate the original array ( which sort() method does! ). So you want to make a copy of `listOfmovies` array, and only then sort it. I shared the link yesterday of several options to clone array